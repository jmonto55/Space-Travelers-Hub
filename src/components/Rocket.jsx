import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { reserveRocket } from '../redux/rockets/rocketsSlice';

const Rocket = ({
  id, name, description, img, reserved,
}) => {
  const dispatch = useDispatch();

  return (
    <div className="flex justify-between items-center h-44" id={id}>
      <img className="w-60 h-full" src={img} alt={name} />
      <div className="flex flex-col justify-between px-5 h-full">
        <h2 className="font-semibold text-xl">{name}</h2>
        <p>
          {reserved && <span className="p-0.5 mr-2 rounded-md text-white bg-teal-500">Reserved</span>}
          {description}
        </p>
        {!reserved && (
        <button
          className="w-32 p-2 mr-2 rounded-md text-white bg-blue-500 hover:bg-blue-600"
          type="button"
          onClick={() => {
            dispatch(reserveRocket(id));
          }}
        >
          Reserve Rocket
        </button>
        )}
        {reserved && (
        <button
          className="w-40 p-2 mr-2 rounded-md text-neutral-500 border-solid border-2 border-neutral-500 hover:bg-neutral-200"
          type="button"
          onClick={() => {
            dispatch(reserveRocket(id));
          }}
        >
          Cancel Reservation
        </button>
        )}
      </div>
    </div>
  );
};

Rocket.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Rocket;
