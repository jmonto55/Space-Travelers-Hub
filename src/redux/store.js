import { configureStore, combineReducers } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionReducer from './missions/missionsSlice';

const rootReducer = combineReducers({ rockets: rocketsReducer, missions: missionReducer });

const store = configureStore({ reducer: rootReducer });

export default store;
