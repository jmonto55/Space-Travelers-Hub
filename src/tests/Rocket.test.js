import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Rocket from '../components/Rocket';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Test suite for the Rocket component', () => {
  it('Should render a Snapshot of the Rocket component', () => {
    const rocket = render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );
    expect(rocket).toMatchSnapshot();
  });

  test('Should test if the text "Reserve Rocket" is inside the Rocket component', () => {
    render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );
    expect(screen.getByText('Reserve Rocket')).toBeInTheDocument();
  });
});
