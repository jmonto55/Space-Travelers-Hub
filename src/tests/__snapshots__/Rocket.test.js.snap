// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Rocket component Should render a Snapshot of the Rocket component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-start items-center h-44"
        data-testid="main_container"
        id="id"
      >
        <img
          alt="name"
          class="w-60 h-full"
          src="img"
        />
        <div
          class="flex flex-col justify-between px-5 h-full"
        >
          <h2
            class="font-semibold text-xl"
          >
            name
          </h2>
          <p>
            des
          </p>
          <button
            class="w-36 p-2 mr-2 rounded-md text-white bg-blue-500 hover:bg-blue-600"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-start items-center h-44"
      data-testid="main_container"
      id="id"
    >
      <img
        alt="name"
        class="w-60 h-full"
        src="img"
      />
      <div
        class="flex flex-col justify-between px-5 h-full"
      >
        <h2
          class="font-semibold text-xl"
        >
          name
        </h2>
        <p>
          des
        </p>
        <button
          class="w-36 p-2 mr-2 rounded-md text-white bg-blue-500 hover:bg-blue-600"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
